src/model/createModel.ts

```diff
@@ -38,40 +41,45 @@ export function createModel<
     [InternalEntityProperty.primaryKey]: primaryKey,
   }
 
-  const resolvedProperties = properties.reduce<Record<string, any>>(
-    (entity, property) => {
-      const exactValue = initialValues[property]
-      const propertyDefinition = definition[property]
-
-      log(
-        `property definition for "${modelName}.${property}"`,
-        propertyDefinition,
-      )
+  const resolveProperty = (
+    entity: { [key: string]: unknown },
+    property: string,
+  ) => {
+    const exactValue = get(initialValues, property)
+    const propertyDefinition = get(definition, property)
 
-      // Ignore relational properties at this stage.
-      if ('kind' in propertyDefinition) {
-        return entity
-      }
+    // Ignore relational properties at this stage.
+    if ('kind' in propertyDefinition) {
+      return entity
+    }
 
-      if ('isPrimaryKey' in propertyDefinition) {
-        entity[property] = exactValue || propertyDefinition.getValue()
-        return entity
-      }
+    if ('isPrimaryKey' in propertyDefinition) {
+      set(entity, property, exactValue || propertyDefinition.getValue())
+      return entity
+    }
 
-      if (
-        typeof exactValue === 'string' ||
-        typeof exactValue === 'number' ||
-        typeof exactValue === 'boolean' ||
-        exactValue?.constructor.name === 'Date'
-      ) {
-        log(`"${modelName}.${property}" has a plain initial value:`, exactValue)
-        entity[property] = exactValue
-        return entity
-      }
+    if (
+      typeof exactValue === 'string' ||
+      typeof exactValue === 'number' ||
+      typeof exactValue === 'boolean' ||
+      exactValue?.constructor.name === 'Date' ||
+      Array.isArray(exactValue)
+    ) {
+      log(`"${modelName}.${property}" has a plain initial value:`, exactValue)
+      set(entity, property, exactValue)
+      return entity
+    }
 
-      entity[property] = propertyDefinition()
+    if (isFunction(propertyDefinition)) {
+      set(entity, property, propertyDefinition())
       return entity
-    },
+    }
+
+    return entity
+  }
+
+  const resolvedProperties = properties.reduce<Record<string, any>>(
+    resolveProperty,
```

Comment:
Why separating the reducer into its own function? Is it a purely aesthetic change?

Reply:
I found it more readable, we can revert it if needed

Reply:
I'd say that keeping line changes scoped to the feature is always a good idea. We may do changes like reducer functions separation in a standalone pull request aimed just at that. What are your thoughts on this?

Reply:
That makes sense. Will revert it :)